masu/api/status.py

```diff
@@ -20,42 +20,87 @@
 import logging
 import os
 import platform
+import socket
 import subprocess
 import sys
 
-from flask import jsonify
+from flask import jsonify, request
 
 from masu.api import API_VERSION
+from masu.celery import celery as celery_app
 from masu.config import Config
 from masu.external.date_accessor import DateAccessor
 from masu.util.blueprint import application_route
 
-logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+LOG = logging.getLogger(__name__)
 
 API_V1_ROUTES = {}
 
 
 @application_route('/status/', API_V1_ROUTES, methods=('GET',))
 def get_status():
     """Packages response for class-based view."""
-    status = ApplicationStatus()
+    if 'liveness' in request.args:
+        return jsonify({'alive': True})
+
+    app_status = ApplicationStatus()
     response = {
-        'commit': status.commit,
-        'python_version': status.python_version,
-        'platform_info': status.platform_info,
-        'modules': status.modules,
-        'api_version': status.api_version,
-        'current_datetime': status.current_datetime,
-        'debug': status.debug
+        'celery_status': app_status.celery_status,
+        'commit': app_status.commit,
+        'python_version': app_status.python_version,
+        'platform_info': app_status.platform_info,
+        'modules': app_status.modules,
+        'api_version': app_status.api_version,
+        'current_datetime': app_status.current_datetime,
+        'debug': app_status.debug
     }
     return jsonify(response)
 
 
-# pylint: disable=too-few-public-methods
+# pylint: disable=too-few-public-methods, no-self-use
 class ApplicationStatus():
     """A view that returns status JSON."""
 
-    def _get_commit(self):  # pylint: disable=no-self-use
+    api_version = API_VERSION
+
+    @property
+    def celery_status(self):
+        """Determine the status of our connection to Celery.
+
+        :returns: dict of celery status
+        """
+        state = celery_app.events.State()
+        events = {}
+
+        def announce_worker_event(event):
+            """Announce worker events."""
+            state.event(event)
+            LOG.debug(f'EVENT: {event}')
+            events[event.pop('hostname')] = event
+
+        with celery_app.connection() as connection:
+            recv = celery_app.events.Receiver(connection, handlers={
+                'worker-offline': announce_worker_event,
```

Comment:
I couldn't figure out how to get the `worker-offline` messages to print but this might be a good candidate to log as an error level some time down the road.  I tried stopping a worker while `masu` was running and didn't see any log message.  In fact,  I only saw `worker-heartbeat` when hitting the status end point.

Since this change would require further changes I'm fine with doing this at a later time if we think it would be valuable.  At the end of the day it would be great to have an error log message when we lose workers.

Reply:
100% agreed. This isn't a perfect or complete implementation by any means. 

The `recv.capture()` call has the details about what it's really doing - we send a message for all workers to wake up; then we listen for 5 seconds for no more than 5 messages; then we report on what was received. So, as long as some workers respond in the time allowed, this check is satisfied.

The exceptions it catches are _broker_ connection issues, not worker connection issues.

So, while yes, we're configured to receive messages other than heartbeats, right now it's highly unlikely we'll see any of those messages unless a worker goes up/down within that 5 second window that the receiver is listening.